R/module_nested_tabs.R

```diff
@@ -285,45 +283,33 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, is_module_specifi
 #'
 #' @param module (`teal_module`) module where needed filters are taken from
 #' @param datasets (`FilteredData`) object where needed data are taken from
-#' @param trigger_data (`reactiveVal`) to trigger getting the filtered data
-#' @return list of reactive datasets with following attributes:
-#' - `code` (`character`) containing datasets reproducible code.
-#' - `join_keys` (`JoinKeys`) containing relationships between datasets.
-#' - `metadata` (`list`) containing metadata of datasets.
+#' @return A `tdata` object.
 #'
 #' @keywords internal
-.datasets_to_data <- function(module, datasets, trigger_data = reactiveVal(1L)) {
+.datasets_to_data <- function(module, datasets) {
   checkmate::assert_class(module, "teal_module")
   checkmate::assert_class(datasets, "FilteredData")
-  checkmate::assert_class(trigger_data, "reactiveVal")
 
   datanames <- if (is.null(module$datanames)) datasets$datanames() else module$datanames
 
   # list of reactive filtered data
   data <- sapply(
     datanames,
-    function(x) eventReactive(trigger_data(), datasets$get_data(x, filtered = TRUE)),
+    function(x) datasets$get_data(x, filtered = TRUE),
```

Comment:
This `eventReactive(trigger_data(), ...)` is very important for teal. This blocks invisible inactive tabs to invalidate data reactives. Removing this causes slowdown of teal application, because every change in the filter panel causes recalculation in all modules.

Reply:
Don't we lose this separation anyway by putting all datasets into one `qenv`?

Reply:
No, because each teal_module listens to some data related reactives. Request goes up to teal and if we remove this  `eventReactive(trigger_data(), ...)` then all module will be recalculated (because filtered data has changed).
`trigger_data()` is based on module visibility, so if module is not-visible, `trigger_data()` doesn't change.